src/data/actions/assets.js

```diff
@@ -108,3 +108,52 @@ export const toggleLockAsset = (asset, courseDetails) =>
 export const clearAssetsStatus = () =>
   dispatch =>
     dispatch({ type: assetActions.CLEAR_ASSETS_STATUS });
+
+export const uploadingAssets = count => ({
+  type: assetActions.UPLOADING_ASSETS,
+  count,
+});
+
+export const uploadAssetSuccess = (asset, response) => ({
+  type: assetActions.UPLOAD_ASSET_SUCCESS,
+  asset,
+  response,
+});
+
+export const uploadAssetFailure = (file, response) => ({
+  type: assetActions.UPLOAD_ASSET_FAILURE,
+  file,
+  response,
+});
+
+export const uploadAssets = (files, courseDetails) =>
+  (dispatch, getState) => {
+    dispatch(uploadingAssets(files.length));
+    files.forEach(file => {
+      clientApi.postUploadAsset(courseDetails.id, file)
+        .then((response) => {
+          if (response.ok) {
+            response.json().then(data => {
```

Comment:
I really like the `.then` here instead of how we've done it in a few other places where it's on the same level as the original `.then` this kind of makes more sense to me when reading it the order of operations of what's happening